### PR TITLE
updated npm deps

### PR DESCRIPTION
**Pull request by OmgImAlexis**
For reference this is the command to update the packages properly.
It checks if there's outdated deps and then updates and saves them as `devDependencies`.

```
npm outdated --depth=0 | grep -v Package | awk '{print $1}' | xargs -I% npm install %@latest --save-dev
```

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked
- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

**Comment by miigotu** [comment]
Update available: 1.6.9 (current: 1.6.7)
Run npm install -g bower to update.

maybe this?

**Comment by OmgImAlexis** [comment]
That's still not causing the issue though.

Edit: Shouldn't it update since `npm install -g bower` is run when travis starts?

**Comment by miigotu** [comment]
Maybe this? `wanted: {"node":">=0.12"} (current: {"node":"0.10.36","npm":"1.4.28"})` (for many packages)
There are also a few deprecated packages.

**Comment by OmgImAlexis** [comment]
None of that causes the error and that's normal as some of the packages we use have deps that're outdated.

Is there a way to have travis clear the cached files?

**Comment by miigotu** [comment]
I already did clear all of the build caches, didnt help

`
npm ERR! 404 'remy/npm-tree' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it
npm ERR! 404 It was specified as a dependency of 'snyk-resolve-deps'
`

**Comment by OmgImAlexis** [comment]
How come it's not failing one of the travis builds on the other PR though? Sounds like it might be a travis issue somewhere.

It's building fine for me. Is travis using npm 3.5.2?

**Comment by miigotu** [comment]
node -v v0.10.36
npm -v 1.4.28

The other PR is where it started failing, the npm index changed between builds.

**Comment by OmgImAlexis** [comment]
Maybe it's a travis issue then.

https://github.com/travis-ci/travis-ci/issues/5562

**Comment by miigotu** [comment]
@OmgImAlexis I dont understand why the js was this broken all of a sudden, stuff used before it was defined? Is this maybe new stuff jshint was picking up that it didnt with an older version? Is what I did ok?

**Comment by miigotu** [comment]
Anyways, this looks fine too me, I specified node 0.12 to get it working, and just moved the function defs to just before they are used (I assume grunt-contrib-jshint update made that necessary)

I'm going to merge this, feel free to revert if it is wrong, or fix if I did something wrong, but I need the other builds in other PRs working =P

File: .travis.yml [hunk 1]
```diff
@@ -9,7 +9,12 @@ branches:
   except:
     - master
 
+env:
+  - TRAVIS_NODE_VERSION="0.12"
```
Comment (OmgImAlexis) [inline .travis.yml:13]: We should be using node `5.0.0` not `0.12`.

Comment (miigotu) [inline .travis.yml:13]: I'll PR that change and see what happens =P

File: .travis.yml [hunk 2]
```diff
@@ -9,7 +9,12 @@ branches:
...
+
 before_install:
+  - rm -rf ~/.nvm && git clone https://github.com/creationix/nvm.git ~/.nvm && (cd ~/.nvm && git checkout `git describe --abbrev=0 --tags`) && source ~/.nvm/nvm.sh && nvm install $TRAVIS_NODE_VERSION
```
Comment (OmgImAlexis) [inline .travis.yml:16]: If Travis has nvm just do `nvm install v5.0.0 && nvm use v5.0.0` and that should give us the one we want.

File: .travis.yml [hunk 3]
```diff
@@ -9,7 +9,12 @@ branches:
...
+  - npm update
```
Comment (OmgImAlexis) [inline .travis.yml:17]: We also won't need this.

File: .travis.yml (hunk 1 above)
Comment (OmgImAlexis) [inline .travis.yml:13]: I'm assuming it was installed since enterprise is still using `v0.12`.

File: .travis.yml (hunk 3 above)
Comment (miigotu) [inline .travis.yml:17]: Done. More worried about the js changes though tbh xD. I think they are ok, but I definitely am not a pro in js.

Comment (OmgImAlexis) [inline .travis.yml:17]: From the looks if they're okay. Hopefully @labrys can get the private api started soon so we will actually get logs for the js.

Comment (labrys) [inline .travis.yml:17]: @OmgImAlexis We need to put more ideas together on the v2API first.  Decide on route layout,  initial subset of API methods available, etc.

**Pull request merged**